integration-tests/api/__tests__/admin/product.js

```diff
@@ -845,6 +853,45 @@ describe("/admin/products", () => {
         },
       ])
     })
+
+    it("returns a list of products that belongs to the requested sales channels", async () => {
+      const api = useApi()
+
+      const productSalesChannelData = {
+        id: "product-sales-channel-1",
+        title: "test description",
+      }
+
+      const productSalesChannel = await simpleProductFactory(dbConnection, productSalesChannelData)
+      const salesChannel = await simpleSalesChannelFactory(dbConnection, {
+        name: "test name",
+        description: "test description",
+      })
+      await dbConnection.manager.query(`
+        INSERT INTO product_sales_channel VALUES ('${productSalesChannel.id}', '${salesChannel.id}')
+      `)
```

Comment:
nit(non-blocking): could we maybe add a products field to the `simpleSalesChannelFactory` so it becomes this:

```
      const salesChannel = await simpleSalesChannelFactory(dbConnection, {
        name: "test name",
        description: "test description",
        products: [productSalesChannel.id]
      })
```

The hard-coded query is a bit verbose.